pkg/tfbridge/schema.go

```diff
@@ -1663,59 +1690,75 @@ func extractSchemaInputs(state resource.PropertyValue, tfs shim.Schema, ps *Sche
 		}
 		return resource.NewArrayProperty(v)
 	case state.IsObject():
-		var tfflds shim.SchemaMap
-		if tfs != nil {
-			if res, isres := tfs.Elem().(shim.Resource); isres {
-				tfflds = res.Schema()
-			}
-		}
-		var psflds map[string]*SchemaInfo
-		if ps != nil {
-			psflds = ps.Fields
+		obj := state.ObjectValue()
+		if tfflds, ok := shimutil.CastToTypeObject(tfs); ok {
+			return resource.NewProperty(
+				extractSchemaInputsObject(obj, tfflds, ps.Fields),
+			)
 		}
-		isMap := tfflds == nil
 
-		obj := state.ObjectValue()
+		// state does not represent an object, so it represents a map.
+		//
+		// That means that state's keys don't have special semantics.
 		v := make(map[resource.PropertyKey]resource.PropertyValue, len(obj))
+		etfs, eps := elemSchemas(tfs, ps)
 		for k, e := range obj {
-			_, etfs, eps := getInfoFromPulumiName(k, tfflds, psflds, rawNames || shimutil.IsOfTypeMap(tfs))
-			if isInput := isMap || etfs != nil && (etfs.Optional() || etfs.Required()); !isInput {
-				glog.V(9).Infof("skipping '%v' (not an input)", k)
-				continue
-			}
-
-			ev := extractSchemaInputs(e, etfs, eps, rawNames || shimutil.IsOfTypeMap(tfs))
-			if mustSet := isMap || etfs != nil && (etfs.Required() || !isDefaultOrZeroValue(etfs, eps, ev)); !mustSet {
-				glog.V(9).Infof("skipping '%v' (not required + default or zero value)", k)
-				continue
-			}
-			v[k] = ev
+			v[k] = extractSchemaInputs(e, etfs, eps, rawNames)
 		}
+
+		// To match previous behavior, we insert the default key for Map types.
+		//
+		// TODO: We should probably remove the extraneous defaultsKey here.
 		v[defaultsKey] = resource.NewArrayProperty([]resource.PropertyValue{})
```

Comment:
Before this PR, we didn't clearly distinguish between Map and Object very well, so we inserted the defaults list in all cases. To avoid changing a bunch of existing tests, I'm leaving the old behavior as is for now. 

Map's can't have field defaults, so I'm pretty sure that this is wrong. Fixing is a followup PR job.